tests/mysql_test.go

```diff
@@ -120,3 +128,120 @@ func TestCheckServerVersionAndBinlogForPITR(t *testing.T) {
 	err = mysqlRestore.CheckBinlogRowFormat(ctx)
 	a.NoError(err)
 }
+
+func TestFetchBinlogFiles(t *testing.T) {
+	t.Parallel()
+	a := require.New(t)
+	ctx := context.Background()
+	log.SetLevel(zapcore.DebugLevel)
+
+	port := getTestPort(t.Name())
+	_, stopFn := resourcemysql.SetupTestInstance(t, port)
+	defer stopFn()
+
+	db, err := connectTestMySQL(port, "")
+	a.NoError(err)
+	defer db.Close()
+
+	driver, err := getTestMySQLDriver(ctx, strconv.Itoa(port), "")
+	a.NoError(err)
+	defer driver.Close(ctx)
+
+	connCfg := getMySQLConnectionConfig(strconv.Itoa(port), "")
+	mysqlDriver, ok := driver.(*pluginmysql.Driver)
+	a.Equal(true, ok)
+	utilDir := t.TempDir()
+	utilInstance, err := mysqlutil.Install(utilDir)
+	a.NoError(err)
+	binlogDir := t.TempDir()
+	mysqlRestore := restoremysql.New(mysqlDriver, utilInstance, connCfg, binlogDir)
+
+	// init schema
+	_, err = db.ExecContext(ctx, `
+		CREATE DATABASE test;
+		USE test;
+		CREATE TABLE tbl (id int);
+		`)
+	a.NoError(err)
+
+	// Rotate to create multiple binlog files for test.
+	var startTsList []int64
+	numRotate := 10
+	// There'll be `numBinlogFiles` binlog files generated, and the last one contains no actual data.
+	numBinlogFiles := numRotate + 1
+	for i := 0; i < numRotate; i++ {
+		// Record the start event ts of the current binlog file.
+		startTsList = append(startTsList, time.Now().Unix())
+		// Insert some data to grow the current binlog file.
+		_, err = db.ExecContext(ctx, fmt.Sprintf(`
+			USE test;
+			INSERT INTO tbl VALUES (%d);
+			`, i))
+		a.NoError(err)
+		// Make a one-second gap between the two binlog files' events
```

Comment:
And explain why wait for 1 second instead of 2?

Reply:
Sharp eyes... This was intended for the behavior of "fetch up to targetTs". Since we stick to downloading all the binlog files as possible, this is no longer needed.
Removed sleeping code.